### PR TITLE
Added fallback locale to media REST API

### LINKED ISSUE #1896: Media selection: Image titles should be displayed in the content language instead of the user language
bildschirmfoto 2015-12-18 um 10 57 48

The image titles are shown in English (=user language) although the current content language is German and the titles would be available in German. This could be confusing for e.g. Korean content administrators as they have no possibility to see the Korean titles when selecting the images.

@danrot: I discussed this with @berndhep, he also thinks that this should be changed.

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #1896 
| Related issues/PRs | https://github.com/massiveart/husky/pull/621
| License | MIT
| Documentation PR | none

#### What's in this PR?

Add the fallback locale to the REST API of the media, and show this locale in the media selection overlay and the media selection content type.

#### Why?

Because knowing from which locale the media title is coming might be an interesting info for the content manager.

### CODE REVIEW

**Comment by danrot** [comment]
The only open question is how we should display the fallback locale in the list view:

image

**Comment by wachterjohannes** [comment]
@danrot i would go with the badge of the datagrid (see categories). The content should be the displayed language @berndhep

**Comment by danrot** [comment]
I think you are right, completely forgot that we have already implemented that somewhere. Just TBH, you are talking about the blue badge here, right?
image

@berndhep Would that be good for you?

**Comment by danrot** [comment]
I have just implemented it like that, and now I think it would make sense to use this blue badge everywhere :confused: Will clarify with @berndhep

**Comment by danrot** [comment]
Talked to @berndhep, I am using this badge always now.
@wachterjohannes Ready for review.

**Comment by danrot** [comment]
@wachterjohannes Maybe wait a second, just had an idea how to solve a problem more elegant :-)

File: src/Sulu/Bundle/MediaBundle/Api/Media.php [hunk 1]
```diff
@@ -789,36 +795,6 @@ public function getDownloadCounter()
     }
 
     /**
-     * Returns array representation of media.
-     *
-     * @return array
-     */
-    public function toArray()
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: Doesn't seem to be used anywhere.

**Comment by danrot** [comment]
@wachterjohannes Really ready to review now :-)

File: src/Sulu/Bundle/MediaBundle/Api/Media.php [hunk 2]
```diff
@@ -910,4 +886,30 @@ private function getMeta($create = false)
...
+        if ($this->localizedMeta) {
+            return $this->localizedMeta;
+        }
+
+        $metas = $this->getFileVersion()->getMeta();
+        $this->localizedMeta = $metas[0];
+
+        foreach ($metas as $key => $meta) {
+            if ($meta->getLocale() == $this->locale) {
+                $this->localizedMeta = $meta;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: use return here

File: src/Sulu/Bundle/MediaBundle/Api/Media.php [hunk 3]
```diff
@@ -910,4 +886,30 @@ private function getMeta($create = false)
...
+                break;
+            }
+        }
+
+        return $this->localizedMeta;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: return null

**Comment by wachterjohannes** [comment]
@danrot please rebase it before i can test it

File: src/Sulu/Bundle/MediaBundle/Api/Media.php (hunk 3 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: I just omit it, has the same effect, and won't make styleci complain :wink:

File: src/Sulu/Bundle/MediaBundle/Api/Media.php (hunk 2 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: fixed

File: src/Sulu/Bundle/MediaBundle/Api/Media.php (hunk 3 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: The change you've suggested is wrong, because null is not returned here.

**Comment by wachterjohannes** [comment]
image

if you upload a new image in the collection and change the title immediately you get a "badge" to another locale (see screenshot).

**Comment by danrot** [comment]
@wachterjohannes That's happening because the file is uploaded in the wrong language. It uses the user language instead the one selected in the right upper corner. Will fix that on tuesday, don't want to do a fast shot here, since this URL is spread across multiple dropbox instances...

**Comment by wachterjohannes** [comment]
@danrot will you fix this issue in this PR too or in another one?

**Comment by danrot** [comment]
@wachterjohannes Just fixed it in here, since the issue is directly related to this new feature.

**Pull request merged**